ReplayKitExample/BroadcastExtension/ExampleReplayKitAudioCapturer.m

```diff
@@ -125,8 +114,25 @@ dispatch_queue_t ExampleCoreAudioDeviceGetCurrentQueue() {
 #pragma clang diagnostic pop
 }
 
-OSStatus ExampleCoreAudioDeviceRecordCallback(CMSampleBufferRef sampleBuffer) {
-    if (!capturingContext || !capturingContext->deviceContext) {
+OSStatus ExampleCoreAudioDeviceCapturerCallback(ExampleReplayKitAudioCapturer *capturer,
+                                                CMSampleBufferRef sampleBuffer) {
+    CMFormatDescriptionRef formatDescription = CMSampleBufferGetFormatDescription(sampleBuffer);
+    const AudioStreamBasicDescription *asbd = CMAudioFormatDescriptionGetStreamBasicDescription(formatDescription);
+    ExampleAudioContext *context = capturer->_capturingContext;
+
+    // Update the capture format ahead of capture starting (no context needed).
+    TVIAudioFormat *format = capturer->_capturingFormat;
+    if (context &&
+        (asbd->mChannelsPerFrame != context->streamDescription.mChannelsPerFrame || asbd->mSampleRate != context->streamDescription.mSampleRate)) {
+        capturer->_capturingFormat = [[TVIAudioFormat alloc] initWithChannels:asbd->mChannelsPerFrame
+                                                                   sampleRate:asbd->mSampleRate
+                                                              framesPerBuffer:format.framesPerBuffer];
+        context->streamDescription = *asbd;
+        TVIAudioDeviceFormatChanged(context->deviceContext);
+        return noErr;
+    }
+
+    if (!context || !context->deviceContext) {
```

Comment:
move this check up..?

Reply:
Yes good catch, this really simplified the logic.